leshan-client-core/src/main/java/org/eclipse/leshan/client/servers/ServerInfo.java

```diff
@@ -65,7 +66,7 @@ public URI getFullUri() {
     }
 
     public boolean isSecure() {
-        return secureMode != SecurityMode.NO_SEC;
+        return secureMode != SecurityMode.NO_SEC || useOscore == true;
```

Comment:
I know this is a bit strange but here I will not really change this.

Historically this was used to know if we should use unsecured(coap) endpoint or secured(coaps) endpoint.

So maybe the right name is "isCoaps" ?
But anyway I read the code again and maybe this does not make so much sense now, so it would be more wise to not touch it for now.

Reply:
Yes, fair enough, I do not need to change the functionality of isSecure. I see your point that it is more about coap vs. coaps (and OSCORE could apply to both cases). I changed this in commit 17098a79932f4c48aa608b7418edc3119a193be5